src/api.py

```diff
@@ -23,97 +21,136 @@ def home():
+                    #add blob to db
+                    blob_id = f.add_blob(new_blob_info.blob_b64s, cur)
+                    # create dict with new_entry metadata including blob_id
+                    metadata_inst.blob_id = blob_id 
+                    metadata_dict = metadata_inst.dict()
+                    # add metadata entry to db
+                    entry_id = f.add_entry(blob_type, metadata_dict, cur)
+                    api_resp = Response(status_code=200,error_message=None,body={'entry_id':entry_id})
+                    return json.dumps(api_resp.dict())
+                except (TypeError, ValueError) as e:
+                    api_resp = Response(status_code=400, error_message= e, body=None)
+                    return json.dumps(api_resp.dict())
+        except:
+            api_resp = Response(status_code=500, error_message='UnexpectedError', body=None)
+            return json.dumps(api_resp.dict())
+        finally:
+            cur.close()
+            conn.close()
+
+    
+    #TODO catch errors at /update how?
+    @app.route('/update', methods=['POST'])
+    def update():
+        try:
+            conn, cur = db_connect(env)
+            try: 
+                post_data = UpdatePostData.parse_obj(request.get_json())
+                if not f.validate_update_fields(post_data.blob_type, post_data.update_data):
+                    api_resp = Response(status_code= 400,error_message= 'KeyError: invalid blob_type or update fields', body=None)
+                    return json.dumps(api_resp.dict())
+                current_metadata = f.get_current_metadata(post_data.blob_type,post_data.current_entry_id,cur)
+                full_update_dict = f.make_full_update_dict(post_data.update_data, current_metadata)
+                updated_entry_id = f.add_entry(post_data.blob_type, full_update_dict, cur)
+                api_resp = Response(status_code=200, error_message=None, body={'entry_id': updated_entry_id})
+                return json.dumps(api_resp.dict())
+            except (TypeError, ValueError) as e: 
+                    api_resp = Response(status_code=400, error_message= e, body=None)
+                    return json.dumps(api_resp.dict())
+        except:
+            api_resp = Response(status_code=500, error_message='UnexpectedError', body=None)
+            return json.dumps(api_resp.dict())
         finally:
             cur.close()
             conn.close()
-            return json.dumps({'status_code':status_code, 'body':payload})
+            return json.dumps(api_resp.dict())
+
+    @app.route('/fields',methods=['GET'])
+    def get_fields():
+        """
+        Return list of fields for specified blob_type
+        """
+        try:
+            blob_type = request.args.to_dict()['blob_type']
+        except:
+            api_resp = Response(status_code = 400, error_message = 'Blob_TypeError: no blob_type given', body = None)
```

Comment:
for errors like these, it can be helpful to pass back to the client the request body, so they can see the json they sent you. just saves a little time for them when debugging.